community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreNodeRelationshipCursor.java

```diff
@@ -138,7 +163,7 @@ public boolean next()
                     }
 
                     // Type check
-                    if ( !checkType( relationshipRecord.getType() ) )
+                    if ( !isDense && !checkType( relationshipRecord.getType() ) )
```

Comment:
Would be great with an execution path without even the condition for dense nodes.